M68K printer: print the 64-bit cs_m68k_op.imm with PRIx64 (it printed $0 on a 32-bit big-endian host).
Apply to michalsc/capstone at 63f7f3ac:  patch -p1 < this-file

--- a/arch/M68K/M68KInstPrinter.c
+++ b/arch/M68K/M68KInstPrinter.c
@@ -14,6 +14,7 @@
 #define PRIx32 "x"
 #define PRIX32 "X"
 #endif
+#define CS_M68K_IMMX PRIx64
 #include <inttypes.h>
 #include <stdlib.h>
 /* Capstone Disassembly Engine */
@@ -172,8 +173,8 @@
 		case M68K_AM_REGI_ADDR_PRE_DEC: SStream_concat(O, "-(a%d)", (op->reg - M68K_REG_A0)); break;
 		case M68K_AM_REGI_ADDR_DISP: SStream_concat(O, "%s$%x(a%d)", op->mem.disp < 0 ? "-" : "", abs(op->mem.disp), (op->mem.base_reg - M68K_REG_A0)); break;
 		case M68K_AM_PCI_DISP: SStream_concat(O, "$%x(pc)", pc + 2 + op->mem.disp); break;
-		case M68K_AM_ABSOLUTE_DATA_SHORT: SStream_concat(O, "$%x.w", op->imm); break;
-		case M68K_AM_ABSOLUTE_DATA_LONG: SStream_concat(O, "$%x.l", op->imm); break;
+		case M68K_AM_ABSOLUTE_DATA_SHORT: SStream_concat(O, "$%" CS_M68K_IMMX ".w", op->imm); break;
+		case M68K_AM_ABSOLUTE_DATA_LONG: SStream_concat(O, "$%" CS_M68K_IMMX ".l", op->imm); break;
 		case M68K_AM_IMMEDIATE:
 			 if (inst->op_size.type == M68K_SIZE_TYPE_FPU) {
 #if defined(_KERNEL_MODE)
@@ -190,7 +191,7 @@
 				 break;
 #endif
 			 }
-			 SStream_concat(O, "#$%x", op->imm);
+			 SStream_concat(O, "#$%" CS_M68K_IMMX, op->imm);
 			 break;
 		case M68K_AM_PCI_INDEX_8_BIT_DISP:
 			SStream_concat(O, "$%x(pc,%s%s.%c)", pc + 2 + op->mem.disp, s_spacing, getRegName(op->mem.index_reg), op->mem.index_size ? 'l' : 'w');
@@ -305,7 +306,7 @@
 
 	if (MI->Opcode == M68K_INS_INVALID) {
 		if (ext->op_count)
-			SStream_concat(O, "dc.w $%x", ext->operands[0].imm);
+			SStream_concat(O, "dc.w $%" CS_M68K_IMMX, ext->operands[0].imm);
 		else
 			SStream_concat(O, "dc.w $<unknown>");
 		return;
